### ISSUE TITLE
Spoken text input on iOS keyboard somewhat broken

### ISSUE THREAD

**Issue by codefrau** [codefrau/SqueakJS#174]
While speaking, iOS generates multiple versions of the recognized text. For example if you say "3+4" it will first generate `Three` and then replace that with `3+4`. We are not properly deleting the `Three` and instead end up with `Three3+4`.

**Comment by LinqLover** [comment]
I bet this is the same issue I observed for autocorrection on Android, e.g., using Gboard (mentioned [here](https://github.com/codefrau/SqueakJS/issues/166#issuecomment-2087749039)).

**Comment by codefrau** [comment]
> I bet this is the same issue I observed for autocorrection on Android, e.g., using Gboard (mentioned [here](https://github.com/codefrau/SqueakJS/issues/166#issuecomment-2087749039)).

Yes that’s likely. I think it’s different from the cursor issue though.

**Comment by LinqLover** [comment]
I wish both just required implementing two similar edge cases in some keyboard event handler functions, but I have not looked into it. :sweat_smile:
